S3 RETL source IAM role authentication (#1051)
rudderlabs/rudderstack-docs: docs/sources/reverse-etl/amazon-s3.mdx

2. From the left panel, go to **Directory** > **Sources** > **Reverse ETL**. Then, select **Amazon S3**.
3. Assign a name to your source and click **Continue**.

### Configuring the connection credentials

Configure the following settings to authenticate RudderStack to access your S3 account: 

- **Connection Mode**: RudderStack provides the following options to connect to S3:
  - **Cross-Account Role (recommended)**: This option lets you connect to S3 through an <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html">IAM access role</a>. To do so, you need to first create an IAM role for RudderStack with the required permissions to access your S3 account. Refer to the <Link to="#creating-the-rudderstack-iam-role">Creating the RudderStack IAM Role for S3</Link> section below for the detailed steps.
  - **Access Key**: This option lets you connect to S3 using your AWS access key ID and secret access key.

<div class="warningBlock">
It is highly recommended to use the <strong>Cross-Account Role</strong> method for connecting to S3 as the <strong>Access Key</strong> method will be deprecated soon.
</div>

> 

Reviewer: Should we add S3 permissions link here? As its relevant here but at a lower position in the doc, can be missed?

      ],
      "Resource": "*"
    }
  ]
}
```

4. Click **Review policy**. On the **Review** page, enter `read-write-app-bucket`.

### Creating the IAM role

After creating the managed policy, follow these steps to create the IAM role for RudderStack.

1. In the left navigation pane, click **Roles** and go to **Create role**.
2. Under **Trusted entity type**, select **AWS account**, as shown:

<img src="../../assets/destinations/aws-role-1.png" alt="Setting up AWS IAM Role for RudderStack" />

3. Select **Another AWS account** and under **Account ID**, enter `422074288268`, the account ID associated with RudderStack.
4. Under **Options** check **Require external ID** and enter your <Link to="/resources/glossary/#workspace-id">workspace ID</Link> as the **External ID**.

<img src="../../assets/destinations/aws-role-2.png" alt="Setting up AWS IAM Role for RudderStack" />

5. Review all the settings carefully and click **Next** to proceed.
6. In the **Permissions** window, select the check box next to the policy you created in the <Link to="#creating-the-policy">Creating the policy</Link> section above.
7. Review all the settings carefully and click **Next** to proceed.

> 8. Enter a unique name for your role. Note that this name **cannot** be distinguished by case. For example, you cannot create a role named `RUDDERSTACK` if `rudderstack` already exists.

Reviewer: ```suggestion
8. Enter a unique name for your role. Note that this name is **case-sensitive**. For example, you cannot create a role named `RUDDERSTACK` if `rudderstack` already exists.
```

<div class="warningBlock">
You cannot edit the name of the role after it has been created.
</div>

9. **Optional**: Enter the description for this role.
10. Click **Create role** to complete the setup.
11. Finally, copy the **ARN** of this newly created role and paste it in the **Role ARN** field in the dashboard settings.

<div class="infoBlock">
Refer to the <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html">AWS IAM tutorial</a> for more information on delegating access across AWS account using IAM roles.
</div>

## S3 permissions

The **minimum** S3 permissions that need to be attached to IAM role or the access keys (depending on your connection method) are listed below:

```json
"Action": [
  "s3:GetObject",
  "s3:ListBucket"
],
```

## FAQ

> ### Where can I obtain the AWS Access Key ID and the AWS Secret Access Key?

Reviewer: Should we just add the reference link as mentioned in the info block without any steps?

> To get the **AWS Access Key ID** and the **Secret Access Key**, follow these steps:

Reviewer: can remove
